packages/client/e2e/analytics.spec.ts

```diff
@@ -0,0 +1,145 @@
+import { expect, Page } from "@playwright/test"
+
+import { test } from "./fixtures"
+import { ElementTimeout, isOpenFin } from "./utils"
+
+const currencyPairs = [
+  "EURUSD",
+  "USDJPY",
+  "GBPUSD",
+  "GBPJPY",
+  "EURJPY",
+  "AUDUSD",
+  "NZDUSD",
+  "EURCAD",
+]
+
+const currencies = ["NZD", "USD", "JPY", "GBP", "EUR", "CAD", "AUD"]
+
+test.describe("Analytics", () => {
+  let analyticsPage: Page
+
+  test.beforeAll(async ({ context, fxPagesRec }, workerInfo) => {
+    if (isOpenFin(workerInfo)) {
+      analyticsPage = fxPagesRec["fx-analytics"]
+    } else {
+      const pages = context.pages()
+      const mainWindow = pages.length > 0 ? pages[0] : await context.newPage()
+
+      await mainWindow.goto(`${process.env.E2E_RTC_WEB_ROOT_URL}`)
+
+      analyticsPage = mainWindow
+    }
+  })
+
+  test.describe("Profit & Loss section", () => {
+    test("Last position amount is displayed in numerical format @smoke", async () => {
+      const lastPositionAmount = await analyticsPage
+        .locator("[data-testid='lastPosition']")
+        .textContent()
+
+      const regexp = RegExp(`[-+,.0-9]`, "g")
+      expect(lastPositionAmount).toMatch(regexp)
+    })
+
+    test("Last position amount is updated periodically", async () => {
+      const lastpositionLocator = analyticsPage.locator(
+        "[data-testid='lastPosition']",
+      )
+      const initiallastPositionAmount = await lastpositionLocator.textContent()
+
+      if (initiallastPositionAmount)
+        await expect(lastpositionLocator).not.toContainText(
+          initiallastPositionAmount,
+          { timeout: ElementTimeout.NORMAL },
+        )
+    })
+
+    test("Correct text color is displayed based on position value", async () => {
+      const initialpositionLocator = analyticsPage.locator(
+        "[data-testid='lastPosition']",
+      )
+      const amount = Number(
```

Comment:
not sure what's possible .. but be careful re amount being `NaN` (non-numeric text) or `0` (for empty string)

Reply:
Good point, I added a check to make the test fail if the value is not truthy

Reply:
ah, these are numbers we are talking about ... so their boolean equivalent is not relevant (and a bit fuzzy/dangerous, tbh)
0 is perfectly valid (falsy)
I think what I want here is a guard/expect to handle NaN (may seem unlikely but that is what we had the other day)
also, I am thinking we need to check for blank text .. as that will convert happily to 0 but is definitely a failure, whereas 0 itself is not.

Reply:
I know you are only reducing to +ve or -ve, but we should still try to use code that renders the actual number - it might be copied later on! So, as we don't do locale variations, we can just strip the comma to get a parseable representation of the number (for fun, I'll try and come up with a locale-sensitive version, in case we ever extend the UI to show French, German etc. number formats ..

Reply:
I updated the test based on your feedback so we handle all cases gracefully, thanks